project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,386 @@
+/*
+    © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define('project_cost_report.ReportAction', function (require) {
+'use strict';
+
+var core = require('web.core');
+var session = require('web.session');
+var Widget = require('web.Widget');
+var ControlPanelMixin = require('web.ControlPanelMixin');
+var session = require('web.session');
+var framework = require('web.framework');
+var crash_manager = require('web.crash_manager');
+var rpc = require("web.rpc");
+
+var QWeb = core.qweb;
+var _t = core._t;
+
+/**
+ * Get a record ID from an event triggered.
+ *
+ * The record ID is expected to be stored in a data-id attribute on the event target.
+ *
+ * @param {jQuery.Event} - the triggered event
+ * @returns {Number} - the record ID
+ */
+function getRecordIdFromEvent(event){
+    var attributeNode = event.currentTarget.attributes['data-id'];
+    return attributeNode ? parseInt(attributeNode.nodeValue) : false;
+}
+
+/**
+ * Get a record name from an event triggered.
+ *
+ * The record name is expected to be stored in a data-name attribute on the event target.
+ *
+ * @param {jQuery.Event} - the triggered event
+ * @returns {String} - the record name
+ */
+function getRecordNameFromEvent(event){
+    var attributeNode = event.currentTarget.attributes['data-name'];
+    return attributeNode ? attributeNode.nodeValue : false;
+}
+
+var ReportAction = Widget.extend(ControlPanelMixin, {
+    events: {
+        'click .o_project_cost_report__analytic_line': 'analyticLineClicked',
+        'click .o_project_cost_report__product_category': 'productCategoryClicked',
+        'click .o_project_cost_report__product_total': 'productTotalClicked',
+        'click .o_project_cost_report__time_category': 'timeCategoryClicked',
+        'click .o_project_cost_report__time_total_hours': 'hourTotalClicked',
+        'click .o_project_cost_report__time_total': 'hourTotalClicked',
+        'click .o_project_cost_report__outsourcing_category': 'outsourcingCategoryClicked',
+        'click .o_project_cost_report__outsourcing_total': 'outsourcingTotalClicked',
+    },
+    init(parent, action) {
+        this._super.apply(this, arguments);
+        this.controllerURL = action.context.url;
+        this.projectId = action.context.active_id || action.params.active_id;
+        this.reportContext = {
+            active_id: this.projectId,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'active_id' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)